src/com/facebook/buck/ide/intellij/model/folders/HasResourcesRoot.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright 2017-present Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. You may obtain
+ * a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.facebook.buck.ide.intellij.model.folders;
+
+import java.nio.file.Path;
+
+public interface HasResourcesRoot {
```

Comment:
On the second thought it's worth moving common functionality that deals with resources root to this class:

```
public class ResourceFolder {
  @Nullable private Path resourcesRoot;

  public Optional<Path> getRelativeOutputPath() {
    ...
  }
}
```

Then you can just derive `JavaResourceFolder` and `JavaTestResourceFolder` from this class.